### ISSUE TITLE
How to train the model for corner detection and text detection

### ISSUE THREAD

**Issue by uvindub** [anhalu/vietnamese_id_card_ocr#1]
I want to know how do you train the model for corner detection and text detection?
For corner detection:
1. Did you use ultralytics yolov8 algorithm?
2. How do you annotate the corners? as boxes or as points? What is the format the annotation output save? which tool use for the annotation?
For text detection?
1. Did you use yolo for it?
2. Did you use the cropped dataset or did you use the same dataset without cropping that you used for corner detection?

**Comment by anhalu** [comment]
Thank you for your interest in my project. 
1. Yes, I use yolov8 
2. I use yolo annotation format (id x y w h). About the label tool, I use https://github.com/HumanSignal/labelImg or https://github.com/vietanhdev/anylabeling or any label tool which support yolo annotation format 
3. Same with number 1 
4. I prepared 3 type of datasets. 
    - a dataset for corner (corner is visible), and use some augmentation (recommend https://github.com/aleju/imgaug) 
    - a dataset for field (field is the box cover the infomation you want to get) 
    - a dataset for text recognition (you can use paddleocr, teseract, ...) 

Basically, The way to handle project is 
input image -> detect corner -> cv2 affine transform -> detect field -> OCR image to text. 
In the detect corner, The goal is to convert the image to the correct format, you can use other techniques like image matching, ... 
Or in the state detect field, you can use some technique of NLP to get infor you want to get, but the efficiency is not high when the amount of data is small. 
if it helpful, plz start the repos, mask the answer and close the issues. Thanks u so much  😁 @uvindub

**Comment by uvindub** [comment]
thanks for the info. if I need more infor further I will  contact you
